packages/@storefront/navigation/src/navigation-header/index.ts

```diff
@@ -8,13 +8,25 @@ class NavigationHeader {
     icons: {},
   };
 
+  init() {
+    this.setIcon();
+  }
+
+  onUpdate() {
+    this.setIcon();
+  }
+
+  setIcon() {
+    this.state = { icon: this.toggleIcon() };
```

Comment:
This should be spreading `state` in case an implementation extends this class or we decide to add more properties to `state`.

Reply:
good call 👍